api/src/main/java/net/starype/quiz/api/game/Question.java

```diff
@@ -0,0 +1,36 @@
+package net.starype.quiz.api.game;
+
+import net.starype.quiz.api.game.player.UUIDHolder;
+
+import java.util.List;
+
+public interface Question extends UUIDHolder {
+
+    List<QuestionTag> getTags();
+
+    default boolean isTagAttached(QuestionTag tag) {
+        return getTags().contains(tag);
+    }
+
+    void registerTag(QuestionTag tag);
+
+    void forgetTag(QuestionTag tag);
+
+    QuestionDifficulty getDifficulty();
+
+    String getQuestion();
+
+    String getCorrectAnswer();
+
+    int getMaxTrialAmount();
```

Comment:
Isn't that supposed to be defined by the game round ? I think it makes more sense. In case we want different max trial counts for different questions, we could rely on their difficulty instead